patch/0080-TMP-mlxsw-i2c-Prevent-transaction-execution-for-spec.patch

```diff
@@ -0,0 +1,126 @@
+From 392127480c91a26dae2449288229d4957213dcc1 Mon Sep 17 00:00:00 2001
+From: Vadim Pasternak <vadimp@nvidia.com>
+Date: Tue, 31 May 2022 10:27:38 +0300
+Subject: [PATCH v4.9 ISSU WA 1/1] TMP: mlxsw: i2c: Prevent transaction
```

Comment:
Why the TMP tag?

Reply:
Is the TMP tag needed (as in, will this bug fix be removed in the future), or is it expected to be permanent?

Reply:
It is expected to be permanent.
We want to mark it as downstream patch.
There are a lot of not up-streamed patches in Sonic kernel, But there is no convention in Sonic  to tag such patches.
"TMP" in patch name could be replaced with "DS" or "Downstream" or with something like that.

Reply:
I would rather have "Sonic-specific" or "Sonic-only" in the patch rather than TMP.

Reply:
"Sonic-specific" sounds better to me. Can we close with this name?
I suggest to use this tag for all new-coming patches (not only from Nvidia), which are not going to be up-streamed.

Reply:
I believe such a tag is unnecessary. The README suggests, that patches are not wanted, that are not going to be upstreamed. The only exception is:

> Platform specific kernel modules which are impossible or very difficult to be built out of kernel tree.

Why were the patches accepted to be in SONiC in the first place. I really thought these were only backports, and it’s not made easy for reviewers to find information about the feature (ISSU), and why it’s not going to be upstreamed.

Reply:
There is no ISSU feature in up-stream driver, so this patch can't be accepted.

Reply:
I agree that there are a number of patches here that should be upstreamed. However, there can still be patches that are not useful to be upstreamed, such as patches that would break the general usage of the Linux kernel, or some hack that really shouldn't be upstreamed.

In this case, since this is functionality that is not supported by the upstream driver, it wouldn't make sense to commit this bug fix for functionality that's not even present in the upstream Linux kernel. If this changes (i.e. ISSU/warmboot/equivalent functionality) becomes available in the upstream kernel, then this should be upstreamed as well.

Reply:
Thank you, but I still do not understand. Does SONiC add a special module/patch to support ISSU in it’s Linux kernel build? Where are those patches? I could apply this patch here (for current SONiC branch) to the upstream Linux kernel source just fine.

    $ git am ~/src/sonic-linux-kernel/patch/0064-TMP-mlxsw-i2c-Prevent-transaction-execution-for.patch # nvidia-stepan/202106-prevent-transaction
    Wende an: TMP: mlxsw: i2c: Prevent transaction execution for special chip states

`MLXSW_CMD_STATUS_FW_ISSU` sounds to me like it’s support in the firmware and independent from the Linux kernel, so should be submitted upstream.

Reply:
Since upstream doesn’t support ISSU, it means nobody can set FW to ISSU and FW will never get MLXSW_CMD_STATUS_FW_ISSU. That’s we it can’t be up-streamed.